portfolio/src/main/webapp/script.js

```diff
@@ -33,76 +33,120 @@ function addRandomFact() {
-    }, {
-      img: 'grass.jpg',
-      headingColor: 'green',
-      headingBgColor: 'rgba(173,255,47,.2)',
-    },
-  ];
-
-  // Update the index of the current background image
-  curBackground = (curBackground + offset) % backgroundImgs.length;
+  // Update the index of the current background image.
+  currentBackgroundIndex = (currentBackgroundIndex + offset) % backgroundImgs.length;
 
-  // Wrap around to the last image if the index is negative
-  if (curBackground < 0) {
-    curBackground += backgroundImgs.length;
+  // Wrap around to the last image if the index is negative.
+  if (currentBackgroundIndex < 0) {
+    currentBackgroundIndex += backgroundImgs.length;
   }
 
-  const newBackground = backgroundImgs[curBackground];
+  const newBackground = backgroundImgs[currentBackgroundIndex];
 
-  // Update the page with the new background image
+  // Update the page with the new background image.
   const html = document.documentElement;
   html.style.backgroundImage = `url(images/${newBackground.img})`;
 
-  // Update the headings to the color theme for the new image
+  // Update the headings to the color theme for the new image.
   updateColorTheme(newBackground.headingColor, newBackground.headingBgColor);
 }
 
 /**
- * Updates the heading color CSS properties
+ * Updates the heading color CSS properties.
  */
 function updateColorTheme(headingColor, headingBgColor) {
   document.body.style.setProperty('--heading-color', headingColor);
   document.body.style.setProperty('--heading-bg-color', headingBgColor);
+}
+
+/**
+ * Fetches the list of comments from the server and adds them to the DOM.
+ */
+function getComments() {
+  const maxComments = document.getElementById('max-comments').value;
+
+  fetch(`/data?max-comments=${maxComments}`).then((response) => response.json()).then((comments) => {
+    const commentsContainer = document.getElementById('comments-container');
+    commentsContainer.innerHTML = '';
+
+    // Create <h4> and <p> elements for each comment's author and text.
+    for (const comment of comments) {
+      const commentElement = document.createElement('div');
+      createTextElement(commentElement, 'h4', comment.author);
+      createTextElement(commentElement, 'p', comment.commentText);
+      commentsContainer.appendChild(commentElement);
+    }
+  });
+}
+
+/**
+ * Creates an html element containing the specified text and
+ * inserts it as the last child of a given parent element.
+ */
+function createTextElement(parentElement, htmlTag, innerText) {
+  const element = document.createElement(htmlTag);
+  element.innerText = innerText;
+  parentElement.appendChild(element);
+  return element;
+}
+
+/**
+ * Deletes all comments from the server and removes them from the page.
+ */
+function deleteAllComments() {
+  fetch('/delete-data', {
+    method: 'POST'
+  }).then((response) => {
+    const commentsContainer = document.getElementById('comments-container');
+    commentsContainer.innerHTML = '';
```

Comment:
This probably isn't necessary since you call getComments which should also do the same.